server/controllers/user.js

```diff
@@ -21,3 +21,23 @@ export async function signup(request, response) {
     });
   }
 }
+
+export async function login(request, response) {
+  try {
+    const {
+      username,
+      password,
+    } = request.body;
+    const returningUser = new User({ username, password });
+    const user = await returningUser.login();
+    response.status(200).header('x-auth-token', user.token).json({
```

Comment:
send the token back to the user as part of the json object

It's when the user makes request that they send to the server as part of their header